File: Yogscart Alpha/Assets/Scripts/Power Ups/DirtBlock.js
#pragma strict

private var parent : Transform;

function Awake()
{
	parent = transform.parent;
}

function OnCollisionEnter(collision : Collision) 
{
	if(collision.transform.GetComponent(kartScript) != null)
	{
		collision.transform.GetComponent(kartScript).SpinOut();
		
		if(parent != null && collision.transform != parent)
			parent.GetComponent(kartScript).HitEnemy();
	}

	if(collision.transform.GetComponent.<Rigidbody>() != null)
	{
		Destroy(this.gameObject);
	}
}

File: Yogscart C#/Assets/Scripts/Power Ups/Dirtblock.cs
﻿using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Dirtblock : DamagingItem
{
    void OnCollisionEnter(Collision collision)
    {
        if (collision.transform.GetComponent<KartMovement>() != null)
        {
            //Spin the Kart Out
            collision.transform.GetComponent<KartMovement>().SpinOut(true);

            //Make Owner Taunt
            DamagingItem di = GetComponent<DamagingItem>();
            if (di.owner != collision.transform.GetComponent<KartMovement>())
                di.owner.DoTaunt();

            //Get rid of the GameObject
            Destroy(gameObject);
        }
        else if (collision.transform.GetComponent<Egg>()) //If hit another Power Up
        {
            Destroy(gameObject);
        }
    }
}
